#! /usr/bin/env python
# Gimbal task configuration

PACKAGE='fm_task_manager'

from task_manager_lib.parameter_generator import *

gen = TaskParameterGenerator()
#       Name                    Type            Description
#       Default    Min   Max
gen.add("altitude",              double_t, 0,  "The UAV altitude [m].",  5.)
gen.add("longitudinal_overlap",  double_t, 0,  "The desired longitudinal overlap.",  0.6, 0., 1.)
gen.add("longitudinal_fov",      double_t, 0,  "The desired longitudinal field of view [rad].",  1.5708, 0., 3.14159)
gen.add("pitch_cmd",             double_t, 0,  "The desired gimbal pitch w.r.t. UAV [rad].", 0.)
gen.add("gimbal_time",           double_t, 0,  "The time to wait for the gimbal to reach the reference angle [s].", 3.)

exit(gen.generate(PACKAGE, "task_server", "Camera"))
